ui/ui-frontend/projects/pastis/src/app/user-actions/save-profile/save-profile.component.ts

```diff
@@ -177,183 +177,135 @@ export class UserActionSaveProfileComponent implements OnInit, OnDestroy {
-                  const profile: Profile = this.noticeService.paNotice(data, profileVersion, true);
-                  if (!this.editProfile) {
-                    // CREER NOTICE PUIS ASSIGNER LE PROFIL A LA NOTICE
-                    this.profile = Object.assign(profile, this.profile);
-                    this.profileDescription = Object.assign(
-                      this.noticeService.profileFromNotice(data, this.editProfile, false),
-                      this.profileDescription,
-                    );
-                  } else {
-                    this.subscriptions.add(
-                      this.fileService.notice.subscribe((value: ProfileDescription) => {
-                        this.profile = Object.assign(profile, value);
-                        this.profileDescription = value;
-                      }),
-                    );
-                  }
-                  // STEP 1 : Create or update Notice
-                  this.savePA();
-                }
-              });
-          } else if (result.action === 'rattachement') {
-            const dataToSendToPopUp = {} as PastisDialogDataCreate;
-            dataToSendToPopUp.titleDialog = this.popupSaveSelectNoticeTitleDialog;
-            dataToSendToPopUp.subTitleDialog = this.popupSaveCreateNoticeSubTitleDialog;
-            dataToSendToPopUp.okLabel = this.popupSaveCreateNoticeOkLabel;
-            dataToSendToPopUp.cancelLabel = this.popupSaveCreateNoticeCancelLabel;
-            dataToSendToPopUp.profileType = this.profileService.profileType;
-            dataToSendToPopUp.profileVersion = this.profileService.profileVersion;
-            const selectNoticeDialog = this.dialog.open(SelectNoticeComponent, {
-              width: '800px',
-              panelClass: 'pastis-popup-modal-box',
-              data: dataToSendToPopUp,
-            });
+    this.subscriptions.add(this.selectSaveOption());
+  }
+
+  private selectSaveOption(): Subscription {
+    return this.dialog
+      .open(SaveProfileOptionsComponent, {
+        width: '800px',
+        panelClass: 'pastis-popup-modal-box',
+        data: {
+          titleDialog: this.popupSaveTitleDialog,
+          subTitleDialog: this.popupSaveSubTitleDialog,
+          width: '800px',
+          height: '800px',
+          okLabel: this.popupSaveOkLabel,
+          cancelLabel: this.popupSaveCancelLabel,
+        },
+      })
+      .afterClosed()
+      .pipe(filter(({ success }) => success))
+      .subscribe((result) => {
+        if (result.action === 'local') {
+          this.subscriptions.add(this.downloadProfiles(true).subscribe());
+        } else if (result.action === 'creation') {
+          this.subscriptions.add(this.create());
+        } else if (result.action === 'rattachement') {
+          this.subscriptions.add(this.attach().subscribe());
+        }
+      });
+  }
+
+  private create(): Subscription {
+    return this.dialog
+      .open(CreateNoticeComponent, {
```

Comment:
```suggestion
      .open<CreateNoticeComponent, PastisDialogDataCreate>(CreateNoticeComponent, {
```
idem. D'ailleurs, il y a un `height` à virer dans `PastisDialogDataCreate` je pense.